### DISCUSSION TITLE
OAuth2 on same window

### DISCUSSION THREAD

**Discussion by niklas-wulff** [pocketbase/pocketbase#3718] [Q&A]
Hey, I am programming a Web-app in javascript. Everything works fine. Except on my smartphone.

Users only allowed to log in with OAuth. 
On my computer, pocketbase opens a new tab, if I click on login. 
When I click on login at my smartphone, nothing happens xD

Can you help me? 

Best Regards,
Niklas Wulff

**Comment by ganigeorgiev** [comment]
Most likely the mobile browser is blocking the popup but I don't know how to help you debug it.

Is it PWA or regular website opened directly in the mobile browser?

If the mobile browser is Safari, see https://github.com/pocketbase/pocketbase/discussions/3563 and https://github.com/pocketbase/pocketbase/discussions/2429#discussioncomment-5943061.

If the above doesn't help, please try to provide some details about your environment and ideally reproducible code sample to avoid the back-and-forth.

**Comment by valiantlynx** [comment]
dont know if it helps but take what you can

i was able to get the the outh happening on the same window. though i could not use the 
`const authData = await pb.collection('users').authWithOAuth2({ provider: 'google' });`
which creates the popup. 
instead i did it manually with 
`const authData = await pb.collection('users').authWithOAuth2Code(...);`

and did manual exchanges with google. 
to stop the popups i had to do an page redirect with 302 code included and no popups appear but the login functionality works:

test the fuctionality here:
https://www.animevariant.co/signup

**Reply by MuslemRahimi** [reply]
I have the same issues.

Can you please provide more info how you done it. 
What is in the (...) brakets
```
 const authData = await pb.collection('users').authWithOAuth2Code(...);
```

**Comment by heyarviind** [comment]
having the same issue in the mobile (in app browser).

**Reply by ganigeorgiev** [reply]
@heyarviind Please open a new discussion with more details about your use case and and example code sampleto understand what do you mean by "having the same issue". See also the 2 linked discussions mentioned in the above comment https://github.com/pocketbase/pocketbase/discussions/3718#discussioncomment-7514962.
